src/expr/src/scalar/func/impls/timestamp.rs

```diff
@@ -73,6 +80,8 @@ sqlfunc!(
 
 sqlfunc!(
     #[sqlname = "timestamp_with_time_zone_to_timestamp"]
+    #[preserves_uniqueness = false]
```

Comment:
I think @mjibson says [here](https://github.com/MaterializeInc/materialize/pull/15476#issuecomment-1282592187) that this should be `true`.